src/gsi/read_prepbufr.f90

```diff
@@ -2133,12 +2145,15 @@ subroutine read_prepbufr(nread,ndata,nodata,infile,obstype,lunout,twindin,sis,&
                        oelev=r10+selev
                     endif
                     if (kx == 280 )then
-                       it29=nint(hdr(8))
-                       if(it29 == 522 .or. it29 == 523 .or. it29 == 531)then
+                       !Prevent integer overflow by nint
+                       if(hdr(8) < huge_i_kind) then
+                          it29=idnint(hdr(8))
+                          if(it29 == 522 .or. it29 == 523 .or. it29 == 531)then
 !                         oelev=r20+selev
-                          oelev=r20
+                             oelev=r20
+                          end if
                        end if
-                    end if
+                    endif
```

Comment:
Nitpick:  we should either use `endif` or `end if`, not both.

Reply:
Fixed.